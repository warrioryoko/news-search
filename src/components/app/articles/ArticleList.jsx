import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleListItems = articles.map(article => (
    <li key={article.title}>
      <Article {...article} />
    </li>
  ));

  return (
    <ul data-testid="articles">
      {articleListItems}
    </ul>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })).isRequired
};

export default ArticleList;
