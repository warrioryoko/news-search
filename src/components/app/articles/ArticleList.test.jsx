import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList.jsx';

const articleTestArray = [
  { title: 'favorite things 1', 
    author: 'Oscar Hammerstein', 
    description: 'Raindrops on roses and whiskers on kittens'
  },
  { title: 'favorite things 2', 
    author: 'Oscar Hammerstein', 
    description: 'Bright copper kettles and warm woolen mittens' 
  },
  { title: 'favorite things 3', 
    author: 'Oscar Hammerstein', 
    description: 'Brown paper packages tied up with strings' 
  },
  { title: 'favorite things 4', 
    author: 'Oscar Hammerstein', 
    description: 'Cream colored ponies and crisp apple strudels' 
  },
  { title: 'favorite things 5', 
    author: 'Oscar Hammerstein', 
    description: 'Door bells and sleigh bells and schnitzel with noodles' 
  },
  { title: 'favorite things 6', 
    author: 'Oscar Hammerstein', 
    description: 'Wild geese that fly with the moon on their wings' 
  },
  { title: 'favorite things 7', 
    author: 'Oscar Hammerstein', 
    description: 'Girls in white dresses with blue satin sashes' 
  },
  { title: 'favorite things 8', 
    author: 'Oscar Hammerstein', 
    description: 'Snowflakes that stay on my nose and eyelashes' 
  },
  { title: 'favorite things 9', 
    author: 'Oscar Hammerstein', 
    description: 'Silver white winters that melt into springs' 
  },
];

describe('Does the ArticleList component render?', () => {
  afterEach(() => cleanup());
  it('should render the ArticleList component', () => {
    const { asFragment } = render(<ArticleList
      articles={articleTestArray}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
