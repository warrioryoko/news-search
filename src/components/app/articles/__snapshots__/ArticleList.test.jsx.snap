// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does the ArticleList component render? should render the ArticleList component 1`] = `
<DocumentFragment>
  <ul
    data-testid="articles"
  >
    <li>
      <h1>
        favorite things 1
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Raindrops on roses and whiskers on kittens
      </p>
    </li>
    <li>
      <h1>
        favorite things 2
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Bright copper kettles and warm woolen mittens
      </p>
    </li>
    <li>
      <h1>
        favorite things 3
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Brown paper packages tied up with strings
      </p>
    </li>
    <li>
      <h1>
        favorite things 4
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Cream colored ponies and crisp apple strudels
      </p>
    </li>
    <li>
      <h1>
        favorite things 5
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Door bells and sleigh bells and schnitzel with noodles
      </p>
    </li>
    <li>
      <h1>
        favorite things 6
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Wild geese that fly with the moon on their wings
      </p>
    </li>
    <li>
      <h1>
        favorite things 7
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Girls in white dresses with blue satin sashes
      </p>
    </li>
    <li>
      <h1>
        favorite things 8
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Snowflakes that stay on my nose and eyelashes
      </p>
    </li>
    <li>
      <h1>
        favorite things 9
      </h1>
      <p>
        Written by: Oscar Hammerstein
      </p>
      <p>
        Article summary: Silver white winters that melt into springs
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
